packages/core/src/components/forms/controls.tsx

```diff
@@ -129,13 +131,43 @@ const Control: React.SFC<IControlInternalProps> = ({
 // Switch
 //
 
-export interface ISwitchProps extends IControlProps {}
+export interface ISwitchProps extends IControlProps {
+    /**
+     * String to display within the switch control component when the switch is active ("on").
+     */
+    internalTextActive?: string;
+
+    /**
+     * String to display within the switch control component when the switch is inactive ("off").
+     */
+    internalTextInactive?: string;
+}
 
 export class Switch extends React.PureComponent<ISwitchProps> {
     public static displayName = `${DISPLAYNAME_PREFIX}.Switch`;
-
+    private switchText = [
+        <div
+            key={`${Classes.SWITCH}-active-text`}
+            className={classNames(`${Classes.CONTROL_INDICATOR}-child`, `${Classes.SWITCH}-active-text`)}
+        >
+            {this.props.internalTextActive}
+        </div>,
+        <div
+            key={`${Classes.SWITCH}-inactive-text`}
+            className={classNames(`${Classes.CONTROL_INDICATOR}-child`, `${Classes.SWITCH}-inactive-text`)}
```

Comment:
`:first-child` and `:last-child` might actually be a friendlier way of styling these elements, and perfectly valid since there's no way to add siblings.